src/edu/csus/ecs/pc2/shadow/RemoteEventFeedMonitor.java

```diff
@@ -530,64 +577,161 @@ public void run() {
+    //                                                System.out.println ("Adding judgement " + judgementID + " for submission " + submissionID + " with judgement " + judgement + " to RemoteJudgements Map");
                                                     synchronized (remoteJudgementsMapLock) {
                                                         getRemoteJudgementsMap().put(judgementID, submissionID + ":" + judgement);
                                                     }
                                                 }
                                             }
-    
+
                                         } catch (Exception e) {
                                             // TODO design error handling reporting (logging?)
-                                            logAndDebugPrint(log, Level.SEVERE, "Exception processing event: " + event, e); 
+                                            logAndDebugPrint(log, Level.SEVERE, "Exception processing event: " + event, e);
+                                        }
+
+                                    } else if ("state".equals(eventType)) {
+
+                                        // State messages look like this (Sample from DOMjudge Event feed Dhaka WF 2022):
+                                        // {"token":"151701","id":null,"type":"state",
+                                        //  "data":{"started":"2022-11-10T10:53:36.000+06:00","ended":null,"frozen":null,"thawed":null,"finalized":null,"end_of_updates":null},"time":"2022-11-10T10:53:36.014+06:00"}
+                                        // {"token":"229483","id":null,"type":"state",
+                                        //  "data":{"started":"2022-11-10T10:53:36.000+06:00","ended":null,"frozen":"2022-11-10T14:53:36.000+06:00","thawed":null,"finalized":null,"end_of_updates":null},"time":"2022-11-10T14:53:36.112+06:00"}
+                                        // {"token":"257177","id":null,"type":"state",
+                                        //  "data":{"started":"2022-11-10T10:53:36.000+06:00","ended":"2022-11-10T15:53:36.000+06:00","frozen":"2022-11-10T14:53:36.000+06:00","thawed":null,"finalized":null,"end_of_updates":null},"time":"2022-11-10T15:53:36.035+06:00"}
+
+                                        try {
+                                            logAndDebugPrint(log, Level.INFO, "Found event of type " + eventType + ": " + event);
+
+                                            // Can set this to indicate we want to put a message in the GUI log
+                                            String statusMessage = null;
+
+                                            //convert state data into a ShadowRunSubmission object
+                                            ShadowStateMessage contestState = createStateMessage((Map<String, Object>) eventMap.get("data"));
```

Comment:
appears that put it into a ShadowStateMessage object, not a ShadowRunSubmission object

Reply:
Fixed in commit https://github.com/pc2ccs/pc2v9/pull/940/commits/fb5d45610571aba4c1e580e7e36bfe256ad6e3ef